src/services/subsocial/evmAddresses/query.ts

```diff
@@ -1,16 +1,35 @@
-import { SubsocialQueryData, createSubsocialQuery } from '@/subsocial-query/subsocial/query'
+import { AccountData } from '@/pages/api/accounts-data'
+import { createQuery, poolQuery } from '@/subsocial-query'
 
-const getLinkedEvmAddress = async (params: SubsocialQueryData<string>): Promise<string> => {
-  const { api: subsocialApi, data } = params
+import axios from 'axios'
 
-  const api = await subsocialApi.blockchain.api
+export async function getAccountsData(addresses: string[]) {
+  const requestedIds = addresses.filter((id) => !!id)
+  if (requestedIds.length === 0) return []
+  const res = await axios.get(
+    '/api/accounts-data?' + requestedIds.map((n) => `addresses=${n}`).join('&')
```

Comment:
GET request has limits, so if you are going to get 50+ addresses per request, rewrite to POST